modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/transaction/TransactionBindingDelegate.java

```diff
@@ -126,7 +126,7 @@ public void invalidate() {
     } finally {
       if (!bound) {
         try {
-          connectionHandler.release();
+          connectionHandler.invalidate();
```

Comment:
this is assuming that the binding failed because of a connectivity issue, but that's not necessarily the case. The binding can fail for a number of reasons, like the transaction being already bound to an incompatible resource type or other things. In that case, you're severing a connection that might be perfectly valid, leading to issues. 

You can only do this if you're certain that a connection problem occurred, in which case, I would challenge if this is the right place to do this.